### PR TITLE
UI: HDS adoption replace <Button> in lib/replication

### PR DESCRIPTION
**Pull request by Monkeychip**
Screenshots of the buttons replaced in PR.

image
image
Note: this is a broken route, I had to manually enter the URL to see it.
image
image

### CODE REVIEW

File: ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs
Comment (Monkeychip) [inline ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs:1]: image
Note: the flow here is a little off, but outside the scope of HDS work. The action onSubmit is a extended action and does not effectively redirect after the modal closes.

File: ui/lib/replication/addon/templates/mode/secondaries/config-edit.hbs
Comment (Monkeychip) [inline ui/lib/replication/addon/templates/mode/secondaries/config-edit.hbs:1]: image

File: ui/lib/replication/addon/templates/mode/secondaries/add.hbs
Comment (Monkeychip) [inline ui/lib/replication/addon/templates/mode/secondaries/add.hbs:1]: image

File: ui/lib/replication/addon/templates/mode/secondaries/config-create.hbs
Comment (Monkeychip) [inline ui/lib/replication/addon/templates/mode/secondaries/config-create.hbs:1]: image
Note: This flow is broken, you can't access it. I modified the URL to get here. Outside the scope of this ticket. But Replication needs some love.

File: ui/lib/replication/addon/templates/components/replication-summary.hbs
Comment (Monkeychip) [inline ui/lib/replication/addon/templates/components/replication-summary.hbs:1]: I was able to handle easy button below, but also on this page are LinkTos structured like the following. My thought is the buttons on this will be handled in the core directory work and then we can remove the LinkTo covering the whole block. Does that sound okay @hellobontempo ? 
```
<LinkTo @route="mode.index" @model="dr" class="link-plain">
            <ReplicationModeSummary @mode="dr" @cluster={{this.cluster}} @tagName="span" />
          </LinkTo>
```

File: ui/app/templates/components/keymgmt/distribute.hbs
Comment (Monkeychip) [inline ui/app/templates/components/keymgmt/distribute.hbs:1]: image
Was asked to tackle as this was missed in a previous HDS button pr.

File: ui/app/templates/components/keymgmt/distribute.hbs [hunk 1]
```diff
@@ -151,24 +151,16 @@
...
-      </div>
-    </div>
+    <hr class="has-background-gray-100" />
+    <Hds::ButtonSet>
+      <Hds::Button
+        @text={{if (or (not @key) this.isNewKey) "Add key" "Distribute key"}}
+        @icon={{if this.saveModel.isRunning "loading"}}
+        type="submit"
+        data-test-secret-save={{true}}
+        disabled={{or this.validationErrorCount this.createDistribution.isRunning}}
```
Comment (hellobontempo) [inline ui/app/templates/components/keymgmt/distribute.hbs:161]: We no longer disable buttons when there are validation errors so this first condition can be removed
```suggestion
        disabled={{this.createDistribution.isRunning}}
```

File: ui/app/templates/components/keymgmt/distribute.hbs [hunk 2]
```diff
@@ -151,24 +151,16 @@
...
-      </div>
-      <div class="control">
-        <Hds::Button @text="Cancel" @color="secondary" {{on "click" @onClose}} />
-      </div>
-    </div>
+    <hr class="has-background-gray-100" />
+    <Hds::ButtonSet>
+      <Hds::Button
+        @text={{if (or (not @key) this.isNewKey) "Add key" "Distribute key"}}
+        @icon={{if this.saveModel.isRunning "loading"}}
```
Comment (hellobontempo) [inline ui/app/templates/components/keymgmt/distribute.hbs:158]: I think this should be?
```suggestion
        @icon={{if this.createDistribution.isRunning "loading"}}
```

File: ui/lib/replication/addon/templates/components/replication-summary.hbs
Comment (hellobontempo) [reply ui/lib/replication/addon/templates/components/replication-summary.hbs:1]: I'm not sure I follow....we don't need to replace the `LinkTo` components? Those are just links (they don't render as buttons)

File: ui/lib/replication/addon/templates/mode/secondaries/add.hbs [hunk 3]
```diff
@@ -45,18 +45,11 @@
...
-    </div>
-    <div class="control">
-      <LinkTo @route="mode.secondaries" @model={{this.replicationMode}} class="button">
-        Cancel
-      </LinkTo>
-    </div>
-  </div>
+  <hr class="has-background-gray-100" />
+  <Hds::ButtonSet>
+    <Hds::Button @text="Generate token" type="submit" data-test-secondary-add={{true}} />
```
Comment (hellobontempo) [inline ui/lib/replication/addon/templates/mode/secondaries/add.hbs:50]:
```suggestion
    <Hds::Button @text="Generate token" type="submit" data-test-secondary-add />
```

File: ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs
Comment (hellobontempo) [reply ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs:1]: It sounds like this was also an issue before with the old confirm action component? Not sure..just took a peek and all of the action shuffling is hard to follow 🥴

**Review by hellobontempo — CHANGES_REQUESTED** [review]
Overall looks good! Just a couple questions, mainly about updating the `keymgmt/distribute.hbs` loading state

File: ui/app/templates/components/keymgmt/distribute.hbs (hunk 2 above)
Comment (Monkeychip) [reply ui/app/templates/components/keymgmt/distribute.hbs:158]: Ahhhh. Why do I miss these things 🙈 Thanks for catching.

File: ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs
Comment (Monkeychip) [reply ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs:1]: Yeah, I think my take away from replication work is replication needs 💜 outside the scope of this PR, but I can walk you through the flow and why I think it's broken.

File: ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs [hunk 4]
```diff
@@ -20,21 +20,20 @@
...
+    @confirmMessage="This will revoke this secondary token."
+    @disabledMessage={{unless this.id "A secondary ID is required perform revocation."}}
+    @onConfirmAction={{action "onSubmit" "revoke-secondary" "primary" (hash id=this.id)}}
+  />
+  <Hds::Button
+    @text="Cancel"
+    @color="secondary"
+    @route="mode.secondaries"
+    @model={{this.replicationMode}}
+    @disabled={{not this.isRevoking}}
```
Comment (hellobontempo) [inline ui/lib/replication/addon/templates/mode/secondaries/revoke.hbs:37]: I think this should be 
```suggestion
    @disabled={{this.isRevoking}}
```

Since before, `Cancel` rendered unless a revoke was in progress. So cancel is only allowed if a revoke is not running. So disabled should `true` if `this.isRevoking`

**Review by hellobontempo — APPROVED** [review]
Nice job! Thanks for tackling those extra buttons and catching the replication summary link to's
